### DISCUSSION TITLE
Using Windows Subsystem for Linux

### DISCUSSION THREAD

**Discussion by megargayu** [microsoft/vscode-cpptools#7982] [Q&A] [answered]
I've installed `g++` on `wsl` (Windows Subsystem for Linux). However, when I added this to the `c_cpp_properties.json` file:
```json
{
  "configurations": [
    {
      "name": "Win32",
      "includePath": ["${workspaceFolder}/**"],
      "defines": ["_DEBUG", "UNICODE", "_UNICODE"],
      "windowsSdkVersion": "10.0.17763.0",
      "compilerPath": "/usr/bin/g++",
      "cStandard": "c17",
      "intelliSenseMode": "linux-gcc-x86",
      "cppStandard": "c++17"
    }
  ],
  "version": 4
}
```
The following error occurs:
```
Cannot find "C:\Users\Ayush\AppData\Local\Packages\CanonicalGroupLimited.UbuntuonWindows_79rhkp1fndgsc\LocalState\rootfs\usr/bin/g++".
```
However, looking in explorer, this path definitely exists. Does anyone know what's going on here?

**Comment by sean-mcmanus** [comment, marked as answer]
We support WSL via using the Remote - WSL extension to connect to the remote WSL instance, which will run the Linux version of our extension instead of the Windows version of our extension.

**Comment by michelleangela** [comment]
Here is also a tutorial to help get started on WSL with the C++ extension: https://code.visualstudio.com/docs/cpp/config-wsl.

**Comment by megargayu** [comment]
@sean-mcmanus Thanks so much for the answer, it really helped. I just can't seem to be able to press the "Mark as answer" button (it won't let me click it).

Edit: I managed to mark it on my phone
